storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh

```diff
@@ -0,0 +1,589 @@
+    cp *.debian.tar.* $WORKDIR/source_deb
+    cp *.changes $WORKDIR/source_deb
+    cp *.dsc $CURDIR/source_deb
+    cp *.orig.tar.gz $CURDIR/source_deb
+    cp *.debian.tar.* $CURDIR/source_deb
+    cp *.changes $CURDIR/source_deb
+    
+}
+
+build_deb(){
+    if [ $DEB = 0 ]
+    then
+        echo "source deb package will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xrmp" ]
+    then
+        echo "It is not possible to build source deb here"
+        exit 1
+    fi
+    for file in 'dsc' 'orig.tar.gz' 'debian.tar.*' 'changes'
+    do
+        get_deb_sources $file
+    done
+    cd $WORKDIR
+
+    echo "DEB_RELEASE=${DEB_RELEASE}" >> percona-xtrabackup-8.0.properties
+    DEBIAN_VERSION="$(lsb_release -sc)"
+    DSC=$(basename $(find . -name '*.dsc' | sort | tail -n 1))
+    DIRNAME=$(echo $DSC | sed -e 's:_:-:g' | awk -F'-' '{print $1"-"$2"-"$3"-"$4}')
+    VERSION=$(echo $DSC | sed -e 's:_:-:g' | awk -F'-' '{print $4}')
+    ARCH=$(uname -m)
+
+    dpkg-source -x $DSC
+    cd $DIRNAME
+    dch -m -D "$DEBIAN_VERSION" --force-distribution -v "$VERSION-$DEB_RELEASE.$DEBIAN_VERSION" 'Update distribution'
+    dpkg-buildpackage -rfakeroot -uc -us -b
+
+    cd ${WORKDIR}
+    mkdir -p $CURDIR/deb
+    mkdir -p $WORKDIR/deb
+    cp $WORKDIR/*.deb $WORKDIR/deb
+    cp $WORKDIR/*.deb $CURDIR/deb
+}
+
+build_tarball(){
+    if [ $TARBALL = 0 ]
+    then
+        echo "Binary tarball will not be created"
+        return;
+    fi
+    get_tar "source_tarball"
+    cd $WORKDIR
+    TARFILE=$(basename $(find . -name 'percona-xtrabackup-*.tar.gz' | sort | tail -n1))
+    if [ -f /etc/debian_version ]; then
+        export OS_RELEASE="$(lsb_release -sc)"
+    elif [ -f /etc/redhat-release ]; then
+        export OS_RELEASE="centos$(lsb_release -sr | awk -F'.' '{print $1}')"
+        RHEL=$(rpm --eval %rhel)
+        if [ "${RHEL}" -eq 8 ]; then
+            export PATH="/usr/local/bin:$PATH"
+        elif [ "${RHEL}" -eq 7 ]; then
+            source /opt/rh/devtoolset-7/enable
+        elif [ "${RHEL}" -eq 6 ]; then
+            source /opt/rh/devtoolset-7/enable
+            source /opt/rh/rh-python36/enable
+        fi
+    fi
+    #
+    ARCH=$(uname -m 2>/dev/null||true)
+    NAME=$(echo ${TARFILE}| awk -F '-' '{print $1"-"$2}')
+    VERSION=$(echo ${TARFILE}| awk -F '-' '{print $3}')
+    #
+    SHORTVER=$(echo ${VERSION} | awk -F '.' '{print $1"."$2}')
+    TMPREL=$(echo ${TARFILE}| awk -F '-' '{print $4}')
+    RELEASE=${TMPREL%.tar.gz}
+    if [ -f /etc/redhat-release ]; then
+        GCRYPT_VER_TMP=$(yum list installed|grep -i libgcrypt | grep -v dev|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+        export GCRYPT_VER=".libgcrypt${GCRYPT_VER_TMP}"
+    elif [ -f /etc/debian_version ]; then
+        GCRYPT_VER_TMP=$(dpkg -l|grep -i libgcrypt | grep -v dev | awk '{print $2}' | awk -F':' '{print $1}')
+        export GCRYPT_VER=".${GCRYPT_VER_TMP}"
+    fi
```

Comment:
this block can be improved based on the info from get_system subroutine

Reply:
Done